eng/testing/tests.mobile.targets

```diff
@@ -113,15 +113,25 @@
   <Target Condition="'$(TargetOS)' == 'Browser'" Name="BundleTestWasmApp">
     <ItemGroup>
       <AssemblySearchPaths Include="$(PublishDir)"/>
+      <WasmFilesToIncludeInFileSystem Include="@(ContentWithTargetPath)" />
+
+      <!-- Extensibility point to specify binaries or assemblies that are built as part of the test assembly and that we read the files during tests
+      i.e System.Reflection.Metadata.Tests opens itself and inspects metadata using PEReader. -->
```

Comment:
Can you point me to where this happens in the SRM tests? I couldn't quickly find the spot.

Maybe it'd make sense to change the test to read a test assembly from the test assets instead rather than itself?

Reply:
https://github.com/dotnet/runtime/blob/0b25063ec056b0d6668bf0848d108d8c873f3fce/src/libraries/System.Reflection.Metadata/tests/Metadata/Decoding/SignatureDecoderTests.cs#L250

Reply:
@safern it's not clear to me why that would need the assembly as a content file? shouldn't it resolve the assembly from the BCL location?

Reply:
No, because it is trying to do `File.OpenRead("/System.Reflection.Metadata.Tests.dll")` that is what `.Assembly.Location` is returning -- unless there is a bug on that API.

Reply:
Ah right. I think just adding `System.Reflection.Metadata.Tests.dll` won't be enough though since other tests load other assemblies like `System.Private.CoreLib.dll`, e.g. https://github.com/dotnet/runtime/blob/0b25063ec056b0d6668bf0848d108d8c873f3fce/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs#L669-L680

---

I assume this failed for you as well?

In fact I grepped across a local file I have with all test failures in all test assemblies and there are about 600 failures due to `System.IO.FileNotFoundException : Could not find file '/<assembly>.dll`.

We'll probably need to find a better solution rather than embedding all the assemblies as content files again. I'll bring it up in our next wasm sync.

Reply:
As we discussed offline I removed this workaround and I'll wait for the wasm sync discussion and fix these cases as a separate change.